File: packages/distant_core/src/implementations/document.rs
use crate::document::Document;

impl Document {
    fn new() -> Self {
        Document {
            uuid: "".to_string(),
            title: "".to_string(),
            description: "".to_string(),
            citation: "".to_string(),
            publication_date: "".to_string(),
            creators: vec![],
            source_id: "".to_string(),
            source_id_type: "".to_string(),
            source_url: "".to_string(),
            archive_location: "".to_string(),
            files: vec![],
            pages: "".to_string(),
            modified: 0,
            created: 0
        }
    }
}

// impl from Source for Document
impl From<&distant_es_client::responses::search_result::Source> for Document {
    fn from(source: &distant_es_client::responses::search_result::Source) -> Self {
        Document {
            ..Default::default()
        }
    }
}

File: packages/act/src/app/distant_api/models/document.ts
/* tslint:disable */
/* eslint-disable */
import { File } from './file';
import { Person } from './person';

/**
 * / An abstract document.
 */
export interface Document {

  /**
   * / archive location
   */
  archive_location: string;

  /**
   * / citation information
   */
  citation: string;

  /**
   * / created date
   */
  created: number;
  creators: Array<Person>;

  /**
   * / description
   */
  description: string;
  files: Array<File>;

  /**
   * / modified date using Unix timestamp
   */
  modified: number;

  /**
   * pages protobuf number type
   */
  pages: string;

  /**
   * / publication date
   */
  publication_date: string;

  /**
   * / In the case of a Zotero Item, this is the Zotero URI.
   */
  source_id: string;

  /**
   * / Describes what the unique Id refers to.
   */
  source_id_type: string;

  /**
   * / source url, e.g. a direct link to the source, e.g. the DOI of a book or journal
   */
  source_url: string;
  title: string;

  /**
   * / Uuid of the source.
   */
  uuid: string;
}
